ddp-workspace/projects/ddp-dsm-ui/src/app/dashboard-statistics/components/date-range/date-range.component.ts

```diff
@@ -0,0 +1,137 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  OnDestroy,
+  OnInit,
+  Output
+} from '@angular/core';
+import {AbstractControl, FormControl, FormGroup, ValidationErrors, Validators} from '@angular/forms';
+import {DateRangeModel} from '../../models/DateRange.model';
+import {Subject} from 'rxjs';
+import {takeUntil, auditTime} from 'rxjs/operators';
+import {DatePipe} from '@angular/common';
+import {Global} from '../../../globals/globals';
+
+@Component({
+  selector: 'app-date-range',
+  templateUrl: 'date-range.component.html',
+  styleUrls: ['date-range.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class DateRangeComponent implements OnInit, OnDestroy {
+  /**
+   * @MAIN Main Date Range Object
+   */
+  public readonly dateRangeForm: FormGroup = this.generateFormGroup;
+
+  private readonly destroyed$: Subject<void> = new Subject<void>();
+
+  constructor(private datePipe: DatePipe) {
+  }
+
+  @Input('activeDates') activeDatesOnInit: DateRangeModel;
+
+  @Input('disabled') set disabledState(isDisabled: boolean) {
+    typeof isDisabled === 'boolean' && this.disableOrEnable(isDisabled);
+  }
```

Comment:
Do you really need check if `typeof isDisabled === 'boolean'` ?

Reply:
I believe 
```
 @Input('disabled') set disabledState(isDisabled: boolean) {
     const options = {emitEvent: false};
     isDisabled? this.dateRangeForm.disable(options) : this.dateRangeForm.enable(options);   
  }
```

would be enough here

Reply:
Oh, a better. Thanks. Changed!